File: backend/src/entities/Claim.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn } from 'typeorm';
import { Student } from './Student';

@Entity('claims')
export class Claim {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  student_id: number;

  @Column({ type: 'date' })
  date_opened: Date;

  @Column({ type: 'enum', enum: ['late_threshold', 'third_yom_lo_ba_li', 'other'] })
  reason: 'late_threshold' | 'third_yom_lo_ba_li' | 'other';

  @Column({ type: 'json' })
  notified_to: string[]; // ['manager', 'student', 'court_chair']

  @Column({ type: 'enum', enum: ['open', 'closed'], default: 'open' })
  status: 'open' | 'closed';

  @ManyToOne(() => Student, student => student.claims)
  @JoinColumn({ name: 'student_id' })
  student: Student;
}

File: backend/app/models/claim.py
from sqlalchemy import Column, Integer, String, Date, ForeignKey, Enum, JSON
from sqlalchemy.orm import relationship
from .base import Base
import enum

class ClaimReason(enum.Enum):
    LATE_THRESHOLD = "late_threshold"
    THIRD_YOM_LO_BA_LI = "third_yom_lo_ba_li"
    OTHER = "other"

class ClaimStatus(enum.Enum):
    OPEN = "open"
    CLOSED = "closed"

class Claim(Base):
    __tablename__ = "claims"
    
    id = Column(Integer, primary_key=True, index=True)
    student_id = Column(Integer, ForeignKey("students.id"), nullable=False)
    date_opened = Column(Date, nullable=False, index=True)
    reason = Column(Enum(ClaimReason), nullable=False)
    notified_to = Column(JSON, nullable=False)  # Array: ["manager", "student", "court_chair"]
    status = Column(Enum(ClaimStatus), nullable=False, default=ClaimStatus.OPEN)
    
    # Relationships
    student = relationship("Student", back_populates="claims")